lib/compat.php

```diff
@@ -88,13 +88,48 @@ function gutenberg_shim_fix_api_request_plain_permalinks( $scripts ) {
 
 	buildAjaxOptions = wp.apiRequest.buildAjaxOptions;
 
+	function getStablePath( path ) {
+		var splitted = path.split( '?' );
+		var query = splitted[ 1 ];
+		var base = splitted[ 0 ];
+		if ( ! query ) {
+			return base;
+		}
+
+		// 'b=1&c=2&a=5'
+		return base + '?' + query
+			// [ 'b=1', 'c=2', 'a=5' ]
+			.split( '&' )
+			// [ [ 'b, '1' ], [ 'c', '2' ], [ 'a', '5' ] ]
+			.map( ( entry ) => entry.split( '=' ) )
+			// [ [ 'a', '5' ], [ 'b, '1' ], [ 'c', '2' ] ]
+			.sort( ( a, b ) => a[ 0 ].localeCompare( b[ 0 ] ) )
+			// [ 'a=5', 'b=1', 'c=2' ]
+			.map( ( pair ) => pair.join( '=' ) )
+			// 'a=5&b=1&c=2'
+			.join( '&' );
+	};
 	wp.apiRequest.buildAjaxOptions = function( options ) {
 		if ( 'string' === typeof options.path ) {
 			options.path = options.path.replace( '?', '&' );
 		}
 
 		return buildAjaxOptions.call( wp.apiRequest, options );
 	};
+
+	var previousApiRequest = wp.apiRequest;
+	wp.apiRequest = function( request ) {
+		var method = request.method || 'GET';
+		var path = getStablePath( request.path )
+		if ( 'GET' === method && window._wpAPIDataPreload[ path ] ) {
+			return Promise.resolve( window._wpAPIDataPreload[ path ].body );
```

Comment:
The original `wp.apiRequest` returns a `jQuery.Deferred`, meaning a consumer may expect to be able to call `.done`, which won't work with `Promise.resolve`.